shell/platform/tizen/tizen_embedder_engine.cc

```diff
@@ -138,40 +139,46 @@ bool TizenEmbedderEngine::RunEngine(
-  render_task_runner.struct_size = sizeof(FlutterTaskRunnerDescription);
-  render_task_runner.user_data = render_loop_.get();
-  render_task_runner.runs_task_on_current_thread_callback =
-      [](void* data) -> bool {
-    return static_cast<TizenEventLoop*>(data)->RunsTasksOnCurrentThread();
-  };
-  render_task_runner.post_task_callback =
-      [](FlutterTask task, uint64_t target_time_nanos, void* data) -> void {
-    static_cast<TizenEventLoop*>(data)->PostTask(task, target_time_nanos);
-  };
-  render_task_runner.identifier = kRenderTaskRunnerIdentifier;
-#endif  
-
   FlutterCustomTaskRunners custom_task_runners = {};
   custom_task_runners.struct_size = sizeof(FlutterCustomTaskRunners);
   custom_task_runners.platform_task_runner = &platform_task_runner;
+
 #ifdef TIZEN_RENDERER_EVAS_GL
-  custom_task_runners.render_task_runner = &render_task_runner;
+  if (HasTizenRenderer()) {
+    FlutterTaskRunnerDescription render_task_runner = {};
+    render_task_runner.struct_size = sizeof(FlutterTaskRunnerDescription);
+    render_task_runner.user_data = render_loop_.get();
+    render_task_runner.runs_task_on_current_thread_callback =
+        [](void* data) -> bool {
+      return static_cast<TizenEventLoop*>(data)->RunsTasksOnCurrentThread();
+    };
+    render_task_runner.post_task_callback =
+        [](FlutterTask task, uint64_t target_time_nanos, void* data) -> void {
+      static_cast<TizenEventLoop*>(data)->PostTask(task, target_time_nanos);
+    };
+    render_task_runner.identifier = kRenderTaskRunnerIdentifier;
+    custom_task_runners.render_task_runner = &render_task_runner;
+  }
 #endif
 
   FlutterRendererConfig config = {};
-  config.type = kOpenGL;
-  config.open_gl.struct_size = sizeof(config.open_gl);
-  config.open_gl.make_current = MakeContextCurrent;
-  config.open_gl.make_resource_current = MakeResourceCurrent;
-  config.open_gl.clear_current = ClearContext;
-  config.open_gl.present = Present;
-  config.open_gl.fbo_callback = GetActiveFbo;
-  config.open_gl.surface_transformation = Transformation;
-  config.open_gl.gl_proc_resolver = GlProcResolver;
-  config.open_gl.gl_external_texture_frame_callback = OnAcquireExternalTexture;
+  if (HasTizenRenderer()) {
+    config.type = kOpenGL;
+    config.open_gl.struct_size = sizeof(config.open_gl);
+    config.open_gl.make_current = MakeContextCurrent;
+    config.open_gl.make_resource_current = MakeResourceCurrent;
+    config.open_gl.clear_current = ClearContext;
+    config.open_gl.present = Present;
+    config.open_gl.fbo_callback = GetActiveFbo;
+    config.open_gl.surface_transformation = Transformation;
+    config.open_gl.gl_proc_resolver = GlProcResolver;
+    config.open_gl.gl_external_texture_frame_callback =
+        OnAcquireExternalTexture;
+  } else {
+    config.type = kSoftware;
+    config.software.struct_size = sizeof(config.software);
+    config.software.surface_present_callback = SurfacePresentCallback;
```

Comment:
How about this?
```suggestion
    config.software.surface_present_callback =
        [](void* user_data, const void* allocation, size_t row_bytes,
           size_t height) -> bool { return true; };
```

Reply:
I applied this suggestion, it simplifies code a bit.